sunpy/map/mapbase.py

```diff
@@ -2727,10 +2727,17 @@ def plot(self, *, annotate=True, axes=None, title=True, autoalign=False,
         autoalign : `bool` or `str`, optional
             If other than `False`, the plotting accounts for any difference between the
             WCS of the map and the WCS of the `~astropy.visualization.wcsaxes.WCSAxes`
-            axes (e.g., a difference in rotation angle).  If ``pcolormesh``, this
-            method will use :meth:`~matplotlib.axes.Axes.pcolormesh` instead of the
-            default :meth:`~matplotlib.axes.Axes.imshow`.  Specifying `True` is
-            equivalent to specifying ``pcolormesh``.
+            axes (e.g., a difference in rotation angle).
+
+            - If ``pcolormesh``, this method will use :meth:`~matplotlib.axes.Axes.pcolormesh`
+              instead of the default :meth:`~matplotlib.axes.Axes.imshow`. Specifying `True`
+              is equivalent to specifying ``pcolormesh``.
+            - If ``reproject``, the map's data will be automatically reprojected to match the
+              WCS of the target axes using the :meth:`~sunpy.map.GenericMap.reproject_to` method,
+              ensuring proper alignment.
```

Comment:
```suggestion
            - If ``reproject``, the map is reprojected to the axes WCS.  The reprojection extent
              can be explicitly specified through the ``extent`` keyword argument ([left, right,
              bottom, top] in pixel coordinates), otherwise the current axes limits are used.
```